rtt/base/BufferLocked.hpp

```diff
@@ -130,10 +139,14 @@ namespace RTT
                 ++itl;
             }
             // this is in any case the number of elements taken from items.
+            size_t writtenSamples = itl - items.begin(); 
```

Comment:
should be fixed now, how do I get Travis to build ? 
On my local Installation I do net get the Signess Errors, even If I hard add -Wsign-compare,
-Wall is also active. One should not, I use GCC 4.8.4 while travis seems to use 4.6